internal/api/auth.go

```diff
@@ -31,14 +35,19 @@ func AuthenticateMiddleware(authn mgauthn.Authentication, domainCheck bool) func
 				return
 			}
 
+			resp.Type = mgauthn.AccessToken
+			if strings.HasPrefix(token, "pat"+seperator) {
+				resp.Type = mgauthn.PersonalAccessToken
+			}
```

Comment:
This can be done in auth service, During authentication process.  
Is there any reason behind this ?